### ISSUE TITLE
Automatic Uninstaller Should Skip When Uninstaller executable does not exist

### ISSUE THREAD

**Issue by ferventcoder** [chocolatey/choco#1035]
tl;dr: The autouninstaller should see if the uninstaller executable still exists prior to attempting to run it.

Original title: Automatic Uninstaller Fails To Determine Software Uninstalled if it Doesn't Remove the Uninstaller Key

The uninstaller for openvpn doesn't handle the removal of the uninstall registry key, nor does it remove the folder. This makes it a bad uninstaller, but Chocolatey should better be able to handle that.

This makes Chocolatey's auto uninstaller service think it needs to attempt an uninstall because the registry key and the folder for the software still exist. But when it attempts to run the uninstaller, it fails because the uninstaller file no longer exists. 

Reference: 

* [AutomaticUninstallerService.cs showing the checks it performs](https://github.com/chocolatey/choco/blob/e447414d7a49223771a310ef01e9bc24744809ab/src/chocolatey/infrastructure.app/services/AutomaticUninstallerService.cs#L86-L87)
* https://gist.github.com/wget/e0d3c386fc87c7a5685b58f1f4d1a526#file-gistfile1-txt-L310-L331
* [October 31, 2016 7:05 AM](https://gitter.im/chocolatey/choco?at=5817339d806316005ddba7fc)
* [October 31, 2016 5:51 PM](https://gitter.im/chocolatey/choco?at=5817cafa0e25dbfa1181edd7)
* [November 1, 2016 1:47 PM](https://gitter.im/chocolatey/choco?at=5818e35ac3569a036e45c8ee)
